system-tests/lib/cre/don/don.go

```diff
@@ -82,20 +84,25 @@ func BuildTopology(nodeSetInput []*cretypes.CapabilitiesAwareNodeSet) (*cretypes
 				Value: nodeType,
 			})
 
-			// TODO this will only work with Docker, for CRIB we need a different approach
-			// that will need to be aware of namespace name and node naming pattern
-			host := fmt.Sprintf("%s-node%d", donMetadata.Name, j)
+			// TODO think whether it would make sense for infraDetails to also hold functions that resolve hostnames for various infra and node types
+			// and use it with some default, so that we can easily modify it with little effort
+			host := infra.Host(j, nodeType, donMetadata.Name, infraDetails)
```

Comment:
all in all, this is a bit problematic... ideally, that code shouldn't know how to construct the URLs, especially that in case of CRIB we are planning to build them on CRIB-side and save to a file. We cannot use this file here, though, because we build the topology before we start the DONs. And we need these hostnames to generate configs, which are required to start the nodes... So ideally, maybe this should live in some 3rd module that would later be used by CRIB/this one?